### PR TITLE
Options for importing database

### PR DESCRIPTION
**Pull request by gbradley**
This PR adds the ability to import a database - see https://github.com/mehrancodes/laravel-harbor/issues/98. Interested in any feedback!

---

There are two approaches: importing a SQL file or via a seeder.

### Via SQL

To import a SQL file, set the `FORGE_DB_IMPORT_SQL` environment variable to the path of the SQL file on your server. Note that .gz and .zip formats are also supported.

By default your database will be imported after the site is provisioned. To import the database on each deployment, set `FORGE_DB_IMPORT_ON_DEPLOYMENT` to true. Note that your SQL file must begin with the `DROP TABLE IF EXISTS` statements necessary to reset the database.

### Via seeder

To seed the database after provisioning, use the `FORGE_DB_SEED` variable. If true this runs the default seeder, but you may specify a custom seeder name.

If you wish to seed the database on each deployment, you can instead add Laravel's `php artisan migrate:fresh --seed` command to your deploy script.

---

Limitations / improvements:

- resetting the DB when importing on deploy. This is an awkward one as it forces the user to manually prepend commands to their SQL dump. Perhaps an alternative would be to fully delete and recreate the database each time.

- database must exist on the server. A more flexible approach might be for Harbor to inject an environment variable that indicates if this is the site's first deployment or not; with `auto_source` enabled, users would then be able to detect this value inside their deploy script and handle downloading & importing the database via their own custom script.

### CODE REVIEW

**Comment by mehrancodes** [comment]
@gbradley Thank you so much for this big enhancement! I'll review and test it out as soon as possible.

**Comment by mehrancodes** [comment]
@gbradley Thank you so much for this big enhancement! I'll review and test it out soon.

**Comment by mehrancodes** [comment]
It looks like the Forge SDK doesn't show errors when running commands. On my first attempt, the provision process finished without any issues, and I saw the message "Importing database from /root/database-dump/provisioner.gz." However, the database was still empty. I found out there's a file permission issue when I tried running the same import command manually on the server.

```
$ gunzip < /root/databases-dump/database.gz | mysql -u 325_update_welcome_page -pT7itw.....   325_update_welcome_page

bash: /root/database-dump/provisioner.gz: Permission denied
mysql: [Warning] Using a password on the command line interface can be insecure.
```

**Comment by mehrancodes** [comment]
@gbradley BTW, thanks for your patience on this PR. The changes look pretty good, and I'm really loving the new test assertions! I'll be thoroughly testing it out this week and will let you know about any possible edge cases and errors.

**Comment by mehrancodes** [comment]
✅ Works smoothly when there is no file permission issue

File: app/Services/Forge/Pipeline/ImportDatabaseFromSql.php [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        ]);
+    }
+
+    protected function buildDatabaseConnection(ForgeService $service): string
+    {
+        if (str_contains($service->server->databaseType, 'postgres')) {
+            return sprintf(
+                'pgsql postgres://%s:%s%s/%s',
+                $service->database['DB_USERNAME'],
+                $service->database['DB_PASSWORD'],
```
Comment (mehrancodes) [inline app/Services/Forge/Pipeline/ImportDatabaseFromSql.php:84]: The database property from $service only gets filled on the first run. One option is to refetch the env keys to get the credentials. The TextToArray action class might be useful here

**Review by mehrancodes — APPROVED** [review]

File: app/Services/Forge/Pipeline/ImportDatabaseFromSql.php (hunk 1 above)
Comment (gbradley) [reply app/Services/Forge/Pipeline/ImportDatabaseFromSql.php:84]: Ah, ok. I’ll try to resolve that soon!

On a related note, I discovered that Forge only waits for 2 minutes before marking a command as failed. So if this feature gets merged, we should note that this may not work in larger datasets that take longer to import.

**Comment by mehrancodes** [comment]
Closing this in favor of https://github.com/mehrancodes/laravel-harbor/pull/162.

Harbor has moved to the Saloon-based Forge API client since this PR was opened, so the original branch no longer applies cleanly on `main`. We rebased the feature onto current `main`, resolved the conflicts, and opened #162 with credit to @gbradley (including a `Co-authored-by` trailer on the commit).

Thank you again for the original work and patience on this — please continue any further discussion on #162.
